src/pages/explorer/index.tsx

```diff
@@ -10,17 +15,24 @@ import { ExplorerTabs } from 'components/explorer/ExplorerTabs';
 import { useEffect, useState } from 'react';
 import { explorerEvents } from 'services/events/explorerEvents';
 import { toPairSlug } from 'utils/pairSearch';
-import config from 'config';
+import { lsService } from 'services/localeStorage';
 
 export const ExplorerPage = () => {
   const { slug, type } = useExplorerParams();
   const navigate = useNavigate({ from: '/explore/$type/$slug' });
 
   useEffect(() => {
+    if (slug && type === 'token-pair') {
+      // Set last visited pair
+      const [base, quote] = slug.split('_');
```

Comment:
Consider getting the baseToken and quoteToken from getTokenById like so and check they exist before setting the tradePair in local storage in case of wrong address pair

```typescript
      const [base, quote] = slug.split('_');
      const baseToken = getTokenById(base);
      const quoteToken = getTokenById(quote);
      if (!baseToken || !quoteToken) return;
```